pkg/agent/agent_windows.go

```diff
@@ -245,3 +247,82 @@ func (i *Initializer) restoreHostRoutes() error {
 	}
 	return nil
 }
+
+func getServicePath() (string, error) {
+	p, err := exec.LookPath(os.Args[0])
+	if err != nil {
+		return "", err
+	}
+	return filepath.Abs(p)
+}
+
+// registerUnregisterService is an entrypoint early in the daemon startup
+// to handle (un-)registering against Windows Service Control Manager (SCM).
+// It returns an indication to stop on successful SCM operation, and an error.
+func (i *Initializer) registerServiceforOS() error {
```

Comment:
@tnqn So now that I have moved this to `windows_agent.go` would I need to call this in `agent/agent.go` where the Initializer is defined, but a lot of the code there does openflow related stuff so was a bit confused